### PR TITLE
Update PS kernel return code location to end of arguments

### PR DESCRIPTION
**Pull request by jeffli-xilinx**
#### Problem solved by the commit
Move PS kernel return code after arguments in the command buffer

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Prevents overwrite of PS kernel command buffer

#### How problem was solved, alternative solutions (if any) and why they were rejected
Calculate offset for return code at end of argument list

#### What has been tested and how, request additional testing if necessary
PS kernel validate tests

### CODE REVIEW

**Review by mamin506 — APPROVED** [review]
The change in cu_scu.c looks good to me. Didn't review the skd code.

File: src/runtime_src/core/edge/skd/xrt_skd.cpp [hunk 1]
```diff
@@ -86,6 +86,11 @@ namespace xrt {
     }
     args = xrt_core::xclbin::get_kernel_arguments((char *)buf,prop.size,sk_name);
     num_args = args.size();
+    if(args[num_args-1].type == xrt_core::xclbin::kernel_argument::argtype::global)
+      return_offset = (args[num_args-1].offset+PS_KERNEL_REG_OFFSET+16)/4;
```
Comment (rozumx) [inline src/runtime_src/core/edge/skd/xrt_skd.cpp:90]: Magic numbers.   
Lots happening in this line. A comment for future developers would be helpful.

File: src/runtime_src/core/edge/skd/xrt_skd.cpp [hunk 2]
```diff
@@ -86,6 +86,11 @@ namespace xrt {
...
+    else
+      return_offset = (args[num_args-1].offset+PS_KERNEL_REG_OFFSET+((args[num_args-1].size>4)?8:4))/4;
```
Comment (rozumx) [inline src/runtime_src/core/edge/skd/xrt_skd.cpp:92]: Magic numbers.
Lots happening in this line. A comment for future developers would be helpful.

**Review by rozumx — CHANGES_REQUESTED** [review]
Some obfuscated code needs to be refactored along with adding some comments to the code.

File: src/runtime_src/core/edge/drm/zocl/cu_scu.c [hunk 3]
```diff
@@ -76,13 +77,12 @@ static int scu_peek_credit(void *core)
 static void scu_xgq_start(struct xrt_cu_scu *scu, u32 *data)
 {
 	struct xgq_cmd_start_cuidx *cmd = (struct xgq_cmd_start_cuidx *)data;
-	u32 num_reg = 0;
 	u32 i = 0;
 	u32 *cu_regfile = scu->vaddr;
 
-	num_reg = (cmd->hdr.count - (sizeof(struct xgq_cmd_start_cuidx)
+	scu->num_reg = (cmd->hdr.count - (sizeof(struct xgq_cmd_start_cuidx)
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/cu_scu.c:83]: There is a lot going on on this line.  A comment would be helpful for future developers.

Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/cu_scu.c:83]: How are you guaranteeing that cmd->hdr.count will always be larger then the size of the structure.   If this should occur, then band things are going to happen.

**Review by rozumx — APPROVED** [review]
Overall this looks good.  That said, there are a few opportunity to add some more comments :^).

**Comment by chvamshi-xilinx** [comment]
Hi @jeffli-xilinx  , Can you please resolve the conflicts.

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by salindac** [comment]
retest this please

**Pull request merged**